00 Core/MWSE/mods/tew/AURA/Ambient/Interior/interiorToExterior.lua

```diff
@@ -0,0 +1,146 @@
+
+    debugLog("Searching for eligible doors.")
+
+    for door in cell:iterateReferences(tes3.objectType.door) do
+        if not (door.destination and door.destination.cell.isInterior and door.tempData) then
+            goto nextDoor
+        end
+        local cellId = door.destination.cell.id:lower()
+        local visitedCellData = modData.visitedInteriorCells[cellId]
+        if (visitedCellData) and (visitedCellData.type) and (visitedCellData.lastVisited) then
+            local now = tes3.getSimulationTimestamp(true)
+            local last = visitedCellData.lastVisited
+
+            debugLog("Parsing door destination cell: " .. cellId)
+            debugLog(string.format("Last time visited: %.5f game hours ago.", (now - last)))
+
+            if not door.tempData.tew then door.tempData.tew = {} end
+            door.tempData.tew.interiorType = visitedCellData.type
+            door.tempData.tew.actorCount = visitedCellData.actorCount
+            door.tempData.tew.lastVisited = visitedCellData.lastVisited
+            if not table.find(cellData.exteriorDoors, door) then
+                table.insert(cellData.exteriorDoors, door)
+            end
+        end
+        :: nextDoor ::
+    end
+
+    if #cellData.exteriorDoors > 0 then
+        debugLog("Tracking " .. #cellData.exteriorDoors .. " door(s). Resuming exterior timer.")
+        exteriorTimer:reset()
+    else
+        debugLog("Found none of interest.")
+    end
+    cellLast = cell
+end
+
+local function playExteriorDoors()
+	if table.empty(cellData.exteriorDoors) then return end
+	debugLog("Updating exterior doors.")
+	local playerPos = tes3.player.position:copy()
+	for _, door in pairs(cellData.exteriorDoors) do
+		if door ~= nil and door.destination.cell
+        and playerPos:distance(door.position:copy()) < 800 then
+            local doorTrack = common.getTrackPlaying(door.tempData.tew.track, door)
+            local now = tes3.getSimulationTimestamp(true)
+            local last = door.tempData.tew.lastVisited
+            local interiorType = door.tempData.tew.interiorType
+            local actorCount
+            if (last) and (now - last) >= 72 then
+                actorCount = door.tempData.tew.actorCount
+            else
+                actorCount = common.getActorCount(door.destination.cell)
+            end
+            local isEligible = getEligibleCellType(interiorType, actorCount)
+            local cellId = door.destination.cell.id:lower()
+            if isEligible and not doorTrack then
+                debugLog(string.format("Door destination is eligible, adding sound. | cellId: %s | actorCount: %s", cellId, actorCount))
+                -- Get new track every time we approach a door, for variety
+                -- Unless we want to trade variety for ultra-realism
+                -- Naaa, using the same track is boooorin'
+                local track = sounds.getTrack{
+                    module = "interior",
```

Comment:
I think we should change it to a separate module from the get-go. You're absolutely right that these are just too heavy to be playing all at once. I'll prepare some shorter versions, just need to know the folder structure you want and the types - are we going to use all the interior types here?

Reply:
Same folder structure as `I`. Yeah, all interior types are used.

Reply:
Ok it's done under 2285c307622738d433289ce5dd583fe4772bdb76
Feel free to update `soundBuilder` and all that. It's all a bit rusty (and only one per each type for now) - will expand after merging.

I'm excited! 😁

Reply:
Ok resolved conflicts, should be mergeble now. Let me know if you want to keep pushing here (changing to custom IE sounds) or merge and then change to these.

Reply:
Yep, do merge this. :D I'll add the missing piece after.